src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GenerateVCFFromPosteriors.java

```diff
@@ -0,0 +1,224 @@
+
+    public static final String SAMPLE_DIRECTORY_NAME_FULL_NAME = "sample-name-directory";
+    public static final String SAMPLE_DIRECTORY_NAME_SHORT_NAME = "SN";
+
+    private static final String COMMENT_PREFIX = "@";
+
+    @Argument(
+            doc = "List of paths to chunk directories in a sorted order",
+            fullName = POSTERIOR_CALL_DIRECTORY_FULL_NAME,
+            shortName = POSTERIOR_CALL_DIRECTORY_SHORT_NAME
+    )
+    private List<File> orderedChunkDirectoryList = null;
+
+    @Argument(
+            doc = "Name of the sample directory in the gCNV called chunks (it must be the same across all chunks)",
+            fullName = SAMPLE_DIRECTORY_NAME_FULL_NAME,
+            shortName = SAMPLE_DIRECTORY_NAME_SHORT_NAME
+    )
+    private String sampleDirectoryName = null;
+
+    @Argument(
+            doc = "Output VCF file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+    )
+    private File outputFile = null;
+
+    /**
+     * Sample name derived from the copy number posterior file
+     */
+    private String sampleName;
+
+    final String variantPrefix = "CNV";
+
+    @Override
+    public void traverse() {
+        if (orderedChunkDirectoryList.size() <= 0) {
+            throw new UserException.BadInput("You must specify at least one chunk directory");
+        }
+
+        final SimpleIntervalCollection firstChunkSimpleIntervalCollection =
+                new SimpleIntervalCollection(new File(getIntervalFileFromChunkDirectory(orderedChunkDirectoryList.get(0).getAbsolutePath()).getAbsolutePath()));
+        final SAMSequenceDictionary samSequenceDictionary = firstChunkSimpleIntervalCollection.getMetadata().getSequenceDictionary();
+        Triple<List<CopyNumberPosteriorLocatableRecord>, IntegerCopyNumberStateCollection, String> firstCopyNumberPosteriorFileInfo =
+                readChunkedPosteriorFileFromDirectory(orderedChunkDirectoryList.get(0), samSequenceDictionary);
+        final IntegerCopyNumberStateCollection copyNumberStateCollection = firstCopyNumberPosteriorFileInfo.getMiddle();
+        sampleName = firstCopyNumberPosteriorFileInfo.getRight();
+
+        if (firstCopyNumberPosteriorFileInfo.getMiddle().size() < 3) {
+            throw new UserException.BadInput("There should be at least 3 copy number states");
+        }
+
+        final VariantContextWriter outputWriter = GATKVariantContextUtils.createVCFWriter(outputFile, null, false);
+        final GCNVPostProcessor gcnvPostProcessor = new GCNVPostProcessor(outputWriter, copyNumberStateCollection,
+                sampleName, samSequenceDictionary);
+
+        //TODO pass the command line invocation string to the header composer method
+        gcnvPostProcessor.composeVariantContextHeader(null);
+        int currentChunk = 0;
+
+        for(File chunkRootDirectory: orderedChunkDirectoryList) {
+            currentChunk++;
```

Comment:
you increment this here and at the end of the loop, which throws off your logged count

Reply:
Done